### PR TITLE
Add more prompts

### PR DESCRIPTION
**Pull request by NathanReb**
This PR cleans up a few things, improve the logs in some places and add prompts before critical actions.

### CODE REVIEW

File: bin/opam.ml [hunk 1]
```diff
@@ -191,6 +170,34 @@ let submit ~dry_run ~opam_repo local_repo remote_repo pkgs auto_open =
...
+let submit ~dry_run ~yes ~opam_repo local_repo remote_repo pkgs auto_open =
+  Config.token ~dry_run () >>= fun token ->
+  List.fold_left (fun acc pkg ->
+      get_pkg_dir pkg
+      >>= fun pkg_dir -> Sos.dir_exists ~dry_run pkg_dir
+      >>= function
+      | true  -> acc
+      | false ->
+          Logs.err (fun m ->
+              m "Package@ %a@ does@ not@ exist. Did@ you@ forget@ \
```
Comment (avsm) [inline bin/opam.ml]: I think Fmt takes care of the `@ ` space mapping for you automatically as part of Logs, so you can just use a normal space

File: lib/github.ml [hunk 2]
```diff
@@ -293,17 +298,25 @@ let publish_distrib ~dry_run ~msg ~archive p =
   >>= fun tag -> check_tag ~dry_run vcs tag
   >>= fun () -> dev_repo p
   >>= fun upstr ->
+  Prompt.confirm_or_abort ~yes ~question:(fun l -> l "Push tag %a?" Text.Pp.version tag)
```
Comment (avsm) [inline lib/github.ml]:
```suggestion
  Prompt.confirm_or_abort ~yes ~question:(fun l -> l "Push tag %a to %a?" Text.Pp.version tag Text.Pp.url upstr)
```

include target of push as well

File: bin/opam.ml (hunk 1 above)
Comment (NathanReb) [reply bin/opam.ml]: This change is actually irrelevant to this PR as this line was just moved but I'm happy to remove those as they'll make the whole thing a lot clearer!

Comment (avsm) [reply bin/opam.ml]: glad@ to @ hear@ that! @ :) @ :)

**Comment by samoht** [comment]
Looks good, thanks!

**Pull request merged**